modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/index/contributor/CSDiagramEntryModelDocumentContributor.java

```diff
@@ -39,6 +46,58 @@ public void contribute(Document document, CSDiagramEntry csDiagramEntry) {
 		document.addText(CPField.SKU, csDiagramEntry.getSku());
 		document.addNumber("quantity", csDiagramEntry.getQuantity());
 		document.addText("sequence", csDiagramEntry.getSequence());
+
+		CPDefinition cpDefinition =
+			_cpDefinitionLocalService.fetchCPDefinitionByCProductId(
+				csDiagramEntry.getCProductId());
+
+		if (cpDefinition == null) {
+			return;
+		}
+
+		String cpDefinitionDefaultLanguageId =
+			LocalizationUtil.getDefaultLanguageId(cpDefinition.getName());
+
+		List<String> languageIds =
+			_cpDefinitionLocalService.getCPDefinitionLocalizationLanguageIds(
+				cpDefinition.getCPDefinitionId());
+
+		for (String languageId : languageIds) {
+			String description = cpDefinition.getDescription(languageId);
+			String name = cpDefinition.getName(languageId);
+			String shortDescription = cpDefinition.getShortDescription(
+				languageId);
+
+			if (languageId.equals(cpDefinitionDefaultLanguageId)) {
+				document.addText(Field.DESCRIPTION, description);
+				document.addText(Field.NAME, name);
+				document.addText(CPField.SHORT_DESCRIPTION, shortDescription);
+			}
```

Comment:
hey @dsitu this block and the block below seems to me the same?

Reply:
oh yeah, taking a closer look, they are adding the same thing. I copied this from cpDefinitionIndexer, ill change them both to keep it consistent. Thanks!